### ISSUE TITLE
Web UI craches when reported target account is deleted

### ISSUE THREAD

**Issue by tribela** [glitch-soc/mastodon#1873]
### Steps to reproduce the problem

1. Someone report a remote account
2. Delete remote account
3. Check report notification from web ui
4. crash
...

Error causing line:
app/javascript/flavours/glitch/features/notifications/components/admin_report.js#L84

```js
    const targetAccount = report.get('target_account');
```

Stack trace
-----------

```
TypeError: a is null
render@https://beta.qdon.space/packs/js/flavours/glitch/async/notifications-bfdc66a3f1ed82607f85.chunk.js:1:9252
Uo@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:327314
zo@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:327109
ws@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:362723
ml@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:354166
pl@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:354089
sl@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:351119
Qi/<@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:302760
t.unstable_runWithPriority@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:377816
Ki@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:302469
Qi@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:302707
Yi@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:302640
ll@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:351410
notify@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:109704
notifyNestedSubs@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:110300
i@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:110134
m@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:628467
547/Ue/</</<@https://beta.qdon.space/packs/js/303-7d1ad834db3f09047d35.chunk.js:2:182400
547/Ue/</</<@https://beta.qdon.space/packs/js/303-7d1ad834db3f09047d35.chunk.js:2:182095
547/Ue/</</<@https://beta.qdon.space/packs/js/303-7d1ad834db3f09047d35.chunk.js:2:181878
n/</</<@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:935888
dispatch@https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:629327
47/q/</<@https://beta.qdon.space/packs/js/303-7d1ad834db3f09047d35.chunk.js:2:117249

```

Mapped stack trace
-----------

```
TypeError: a is null
render (webpack:///app/javascript/opt/mastodon/app/javascript/flavours/glitch/features/notifications/components/admin_report.js:84:26)
gi (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4232:7)
Lg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:4225:1589)
ws (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:6536:6)
Bg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5993:12)
Ag (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5988:)
yj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5730:12)
gg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2759:4)
unstable_runWithPriority (webpack:///node_modules/scheduler/cjs/scheduler.production.min.js:292:)
bg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2723:9)
eg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2751:49)
dg (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:2747:12)
Kj (webpack:///node_modules/react-dom/cjs/react-dom.production.min.js:5753:4)
batch(function (webpack:///node_modules/react-redux/es/utils/Subscription.js:15:6)
notifyNestedSubs (webpack:///node_modules/react-redux/es/utils/Subscription.js:85:14)
handleChangeWrapper (webpack:///node_modules/react-redux/es/utils/Subscription.js:90:19)
k (https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:628467)
return (webpack:///app/javascript/opt/mastodon/app/javascript/flavours/glitch/middleware/sounds.js:44:11)
action.type.match (webpack:///app/javascript/opt/mastodon/app/javascript/flavours/glitch/middleware/errors.js:15:11)
action.type.match (webpack:///app/javascript/opt/mastodon/app/javascript/flavours/glitch/middleware/loading_bar.js:23:11)
_createClass (webpack:///node_modules/react-redux-loading-bar/build/loading_bar.js:152:48)
k (https://beta.qdon.space/packs/js/common-02682779b6b8c74752be.chunk.js:2:629327)
compareId (webpack:///app/javascript/opt/mastodon/app/javascript/flavours/glitch/actions/notifications.js:203:6)
```

React component stack
---------------------

```

    in K
    in Connect(K)
    in V
    in Connect(V)
    in article
    in _
    in Connect(_)
    in div
    in div
    in k
    in Connect(k)
    in Connect(k)
    in div
    in s
    in Dt
    in InjectIntl(Dt)
    in Connect(InjectIntl(Dt))
    in c
    in Connect(c)
    in div
    in Wt
    in InjectIntl(Wt)
    in Connect(InjectIntl(Wt))
    in Connect(InjectIntl(Wt))
    in ye
    in div
    in div
    in t
    in t
    in me
    in t
    in withRouter(me)
    in InjectIntl(withRouter(me))
    in Connect(InjectIntl(withRouter(me)))
    in t
    in t
    in t
    in withRouter(t)
    in t
    in t
    in ke
    in c
    in IntlProvider
    in Ae
```

### Expected behaviour

Not crashing

### Actual behaviour

Web UI is crashing

### Specifications

b3030187a41b242697d640f14fb1f589201c55e0

related: https://github.com/mastodon/mastodon/issues/19141

**Comment by jwbjnwolf** [comment]
The temporary solution to this is to hide the "new reports" notifications from being shown in the column.

If your notifications are currently crashing immediately, use an alt or get someone to spam your notifications enough to push that problem report notification so that it's not loaded in until you scroll.

Now open the notification settings on the right side of the tab title bar, and scroll to the bottom and you'll see "new reports". Toggle off the "show in column".

Get each of your admins to do that. If they have the moderator role or a custom role with mod/admin permissions, you may have to temporarily change their role to the stock admin/owner role for them to see the option.

Boom, now you won't have the issue. Though of course, this is still an issue that Glitch needs to sort if this is indeed a glitch specific issue.. though this does happen in both glitch and vanilla flavours.

**Comment by kescherCode** [comment]
This should be fixed by d2eb726962187226c85ef7f2ee1886cb0767bbd1 and ce27c6502bc439acda8ad89a4c5386698dbf82c0 and can be closed.
